submit_cgap/tests/test_submission.py

```diff
@@ -2202,3 +2253,144 @@ def test_running_on_windows_native():
+        ) as mocked_yes_or_no:
+            with mock.patch.object(
+                submission_module,
+                "CGAP_SELECTIVE_UPLOADS",
+                cgap_selective_uploads,
+            ):
+                side_effect = None
+                if error_raised:
+                    side_effect = RuntimeError("Error occurred")
+                simple_function = mock.MagicMock(
+                    side_effect=side_effect, return_value=expected_result
+                )
+
+                uuid = "some_uuid"
+                file_name = "foo/bar"
+                input_arg = "foo"
+                function_wrapper = UploadMessageWrapper(uuid, no_query=no_query)
+                result = function_wrapper.wrap_upload_function(
+                    simple_function, file_name
+                )(input_arg, error_raised=error_raised)
+
+                expected_lines = []
+                if not no_query and cgap_selective_uploads and not yes_or_no_result:
+                    mocked_yes_or_no.assert_called_once()
+                    expected_lines.append("OK, not uploading it.")
+                    simple_function.assert_not_called()
+                else:
+                    expected_lines.append(f"Uploading {file_name} to item {uuid} ...")
+                    simple_function.assert_called_once_with(
+                        input_arg, error_raised=error_raised
+                    )
+                    if error_raised:
+                        expected_lines.append(f"RuntimeError: Error occurred")
+                    else:
+                        expected_lines.append(
+                            f"Upload of {file_name} to item {uuid} was successful."
+                        )
+                assert shown.lines == expected_lines
+                assert result == expected_result
+
+
+@pytest.mark.parametrize(
+    "credentials,files_found,expected_file_search_calls,expected_uploader_calls",
+    [
+        ([], [], 0, 0),
+        ([{"filename": "foo"}], [], 0, 0),
+        ([{"upload_credentials": {"key": "value"}}], [], 0, 0),
+        (SOME_EXTRA_FILE_CREDENTIALS, [], 2, 0),
+        (SOME_EXTRA_FILE_CREDENTIALS, [SOME_FILENAME], 2, 1),
+        (SOME_EXTRA_FILE_CREDENTIALS, [SOME_FILENAME, ANOTHER_FILE_NAME], 2, 2),
+    ]
+)
+def test_upload_extra_files(
+    credentials, files_found, expected_file_search_calls, expected_uploader_calls
+):
+    """Test extra files credentials utilized to search for and then
+    upload files.
+    """
+    folder = SOME_USER_HOMEDIR
+    recursive = True
+    auth = SOME_AUTH
+
+    def mocked_file_search(folder, extra_file_name, **kwargs):
+        if extra_file_name in files_found:
+            return os.path.join(folder, extra_file_name), None
+        else:
+            return None, "error"
+
+    mocked_search_for_file = mock.MagicMock(side_effect=mocked_file_search)
+    with mock.patch.object(
+        submission_module, "search_for_file", mocked_search_for_file
+    ):
```

Comment:
I usually write it this way because there's no need to mention the class MagicMock. It will be created implicitly. YMMV. This is not tested:
```suggestion
    with mock.patch.object(submission_module, "search_for_file") as mock_search_for_file:
        mock_search_for_file.side_effect = mocked_file_search
```